lib/libyui_client/http/libyui_http_client.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module LibyuiClient
+  module Http
+    class LibyuiHttpClient
+      def initialize(app)
+        @app = app
+      end
+
+      def widget_get(filter, timeout: LibyuiClient.timeout, interval: LibyuiClient.interval)
+        res = nil
+        Wait.until(timeout: timeout, interval: interval) do
+          uri = HttpClient.compose_uri(@app.host, @app.port, '/widgets', filter)
+          res = HttpClient.http_get(uri)
+          Response.new(res.code, res.body) if res.code.to_i == 200
+        end
+      rescue Error::TimeoutError
+        rescue_widget_errors(res)
+      end
+
+      def widget_send_action(filter, action, timeout: LibyuiClient.timeout, interval: LibyuiClient.interval)
+        res = nil
+        Wait.until(timeout: timeout, interval: interval) do
+          uri = HttpClient.compose_uri(@app.host, @app.port, '/widgets',
+                                       filter.merge(action))
+          res = HttpClient.http_post(uri)
+          Response.new(res.code, res.body) if res.code.to_i == 200
+        end
+      rescue Error::TimeoutError
+        rescue_widget_errors(res)
+      end
+
+      private
+
+      def rescue_widget_errors(response)
+        raise Error::WidgetNotFoundError if response.code == 404
```

Comment:
Same as above, we see the definitions provided here: https://www.rubydoc.info/stdlib/net/Net/HTTPResponse